src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -609,6 +615,30 @@ bool OnOffServer::offWithEffectCommand(app::CommandHandler * commandObj, const a
     chip::EndpointId endpoint = commandPath.mEndpointId;
     Status status             = Status::Success;
 
+    if (isKnownEnumValue(effectId))
```

Comment:
This could just be a direct compare against `EffectIdentifierEnum::kUnknownEnumValue`, no?  Decoding of a command's fields will check them for validity if they are known enum types, and replace invalid values with `kUnknownEnumValue`

Reply:
True for this enum, but using `isKnownEnumValue` will also compare against end up comparing against `EffectIdentifierEnum::kUnknownEnumValue` and it lines up with the following checks. 

No strong opinion on this as this creates an extra function call. I can change it